cdap-app-fabric/src/main/java/io/cdap/cdap/master/environment/k8s/MasterEnvironmentMain.java

```diff
@@ -110,9 +110,9 @@ public static void doMain(String[] args) throws Exception {
                                                + MasterEnvironmentRunnable.class);
         }
 
-        //TODO: CDAP-17754 Use proper authenticaiton context with internal token from configuration
+        //TODO: CDAP-17754 Use proper authentication context with internal token from configuration
```

Comment:
You can actually remove this comment as I believe you are setting proper context

Reply:
I'm actually not entirely confident that MasterEnvironmentMain should always use WorkerAuthenticationContext. I only noticed that MasterEnvironmentMain is called from preview runner pods. If MasterAuthenticationMain is used anywhere else, it may cause request failures.